File: src/main/java/arriba/senders/VoidSender.java
package arriba.senders;

import java.io.IOException;

import arriba.common.Sender;

/**
 * A sender that only prints the message to be sent.
 * 
 * @param <M>
 */
public final class VoidSender<M> implements Sender<M> {

    public void send(final M message) throws IOException {
        System.out.println(message);
    }
}

File: src/main/scala/arriba/scala/senders/VoidSender.scala
package arriba.scala.senders

import arriba.scala.common.Sender

/**
 * A sender that only prints the message to be sent.
 *
 * @param <M>
 */
class VoidSender[M] extends Sender[M] {
  def send(message: M) {
    println(message)
  }
}

